arm/Microsoft.Insights/metricAlerts/.bicep/nested_rbac.bicep

```diff
@@ -1,11 +1,11 @@
-param roleAssignment object
+param roleAssignmentObj object
 param builtInRoleNames object
 param resourceName string
 
-resource roleAssigment 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = [for principalId in roleAssignment.principalIds: {
-  name: guid(resourceName, principalId, roleAssignment.roleDefinitionIdOrName)
+resource roleAssigment 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = [for principalId in roleAssignmentObj.principalIds: {
```

Comment:
Use the resource full scope

Reply:
Fixed. please check and resolve if you agree.

Reply:
Also name should be updated accordingly
`name: '${resourceName}/Microsoft.Authorization/${guid(resourceName, principalId, roleAssignmentObj.roleDefinitionIdOrName)}'`